### ISSUE TITLE
Error when recovering an entry with references from history

### ISSUE THREAD

**Issue by TheZ3ro** [keepassxreboot/keepassxc#1485]
## Expected Behavior

Recovering an item from history sucessfully

## Current Behavior

Useful strace (from bottom to top)
```
    #0 0x562a3f8becf2 in QWeakPointer<QObject>::data() const /usr/include/x86_64-linux-gnu/qt5/QtCore/qsharedpointer_impl.h:588
    #1 0x562a3f8d710d in QPointer<Database>::data() const /usr/include/x86_64-linux-gnu/qt5/QtCore/qpointer.h:70
    #2 0x562a3f8d5905 in QPointer<Database>::operator Database*() const /usr/include/x86_64-linux-gnu/qt5/QtCore/qpointer.h:76
    #3 0x562a3f8cfd8d in Group::database() /opt/git/keepassxc/src/core/Group.cpp:466
    #4 0x562a3f8b80d1 in Entry::resolveReferencePlaceholderRecursive(QString const&, int) const /opt/git/keepassxc/src/core/Entry.cpp:831
```

Error occours in line [831](https://github.com/keepassxreboot/keepassxc/blob/b80ddef2743f8eb3d952ea1a6f947352ede80d7f/src/core/Entry.cpp#L831)
 `const Entry* refEntry = m_group->database()->resolveEntry(searchText, searchInType);`

## Debug Info

KeePassXC - Versione 2.3.0-snapshot
Build Type: Snapshot
Revisione: 490e921

Librerie:
- Qt 5.5.1
- libgcrypt 1.8.2

Sistema operativo: Ubuntu 16.04.3 LTS
Architettura CPU: x86_64
Kernel: linux 4.13.0-32-generic

Estensioni abilitate:
- Auto-Type
- Browser Integration
- SSH Agent
- YubiKey

**Comment by TheZ3ro** [comment]
Seems that is some cases this works fine, I need to check whats wrong. We can postpone this

**Comment by phoerious** [comment]
Fixed in #1486
